src/LiveComponent/tests/Functional/Util/QueryStringPropsExtractorTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Tests\Functional\Util;
+
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadataFactory;
+use Symfony\UX\LiveComponent\Tests\Fixtures\Dto\Address;
+use Symfony\UX\LiveComponent\Tests\LiveComponentTestHelper;
+use Symfony\UX\LiveComponent\Util\QueryStringPropsExtractor;
+
+class QueryStringPropsExtractorTest extends KernelTestCase
+{
+    use LiveComponentTestHelper;
+
+    /**
+     * @dataProvider getQueryStringTests
+     */
+    public function testExtract(string $queryString, array $expected)
+    {
+        $extractor = new QueryStringPropsExtractor($this->hydrator());
+
+        $request = Request::create('/'.!empty($queryString) ? '?'.$queryString : '');
+
+        /** @var LiveComponentMetadataFactory $metadataFactory */
+        $metadataFactory = self::getContainer()->get('ux.live_component.metadata_factory');
+
+        $metadata = $metadataFactory->getMetadata('component_with_url_bound_props');
+        $component = $this->getComponent('component_with_url_bound_props');
+
+        $data = $extractor->extract($request, $metadata, $component);
+
+        $this->assertEquals($expected, $data);
+    }
+
+    public function getQueryStringTests(): iterable
+    {
+        yield from [
+            ['', []],
+            ['prop1=foo', ['prop1' => 'foo']],
+            ['prop2=42', ['prop2' => 42]],
```

Comment:
There are edge cases i'm not sure how we should handle. 

But if you add

```php
    #[LiveProp(writable: true, url: true)]
    public int $prop6 = 123;

    #[LiveProp(writable: true, url: true)]
    public ?int $prop7 = 123;
```

Then in the tests, i got those cases i'm not sure of

(when i write `--> []` it is the result obtained)

```php
            ['prop6=122', ['prop6' => 122]],
            ['prop6=', []], // NOK --> ['prop6' => 0]
            ['prop6=0', ['prop6' => 0]],
            ['prop6=-1', ['prop6' => -1]],
            ['prop6=00', []], // or exception NOK --> ['prop6' => 0]
            ['prop6=01', []], // or exception NOK --> ['prop6' => 1]
            ['prop6=abc', []], // or exception NOK --> ['prop6' => 0]

            ['prop7=', ['prop7' => null]], // NOK --> []
```

Reply:
But should we not stop here before the type cohercion ? (genuine question)
If not you're right, that is outside this scope :)

Reply:
> But should we not stop here before the type cohercion ? (genuine question)

You mean with an additional check before we call the hydrator, to ensure the coercion will be consistent?